src/device/rsp/rsp_core.c

```diff
@@ -306,27 +304,45 @@ void do_SP_Task(struct rsp_core* sp)
         rsp.doRspCycles(0xffffffff);
         timed_section_end(TIMED_SECTION_AUDIO);
         sp->regs2[SP_PC_REG] |= save_pc;
+
+        sp_delay_time = 4000;
     }
     else
     {
         sp->regs2[SP_PC_REG] &= 0xfff;
         rsp.doRspCycles(0xffffffff);
         sp->regs2[SP_PC_REG] |= save_pc;
+
+        sp_delay_time = 0;
     }
 
     sp->rsp_task_locked = 0;
     if ((sp->regs[SP_STATUS_REG] & (SP_STATUS_HALT | SP_STATUS_BROKE)) == 0)
```

Comment:
oh yeah if you can get rid of the rsp_task_locked flag that would be nice, savestates don't work as you mention. If you figure it out I would submit a different PR though, it's sort of a separate issue from this PR.